pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -270,7 +270,8 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                         // creation request either complete or fails.
                         log.warn("[{}][{}][{}] Consumer is already present on the connection", remoteAddress, topicName,
                                 subscriptionName);
-                        ctx.writeAndFlush(Commands.newError(requestId, ServerError.UnknownError,
+                        ServerError error = !existingConsumerFuture.isDone() ? ServerError.ServiceNotReady : ServerError.UnknownError;
+                        ctx.writeAndFlush(Commands.newError(requestId, error,
```

Comment:
yes, if request is already in process then now, we are returning ```ServiceNotReady``` error instead unknown-error.

if consumer times-out before broker completes the request and if broker receives duplicate request while it is still processing the request then it was returning ```unknown``` error which was preventing client to retry again. 
however, now, broker is returning ```ServiceNotReady``` error which is retriable error and client can retry later on and reconnect again.